react/features/invite/actions.js

```diff
@@ -64,3 +74,85 @@ export function updateDialInNumbers() {
+ * numbers, and video rooms).
+ *
+ * @param  {Array<Object>} invites - Items for which invites should be sent.
+ * @returns {Promise} Promise containing the list of invites that were not sent.
+ */
+export function sendInvitesForItems(invites: Array<Object>) {
+    return (
+            dispatch: Dispatch<*>,
+            getState: Function): Promise<Array<Object>> => {
+        let allInvitePromises = [];
+        let invitesLeftToSend = [ ...invites ];
+        const state = getState();
+        const { conference } = state['features/base/conference'];
+        const { inviteServiceUrl } = state['features/base/config'];
+        const inviteUrl = getInviteURL(state);
+        const jwt = state['features/base/jwt'].jwt;
+
+        // First create all promises for dialing out.
+        if (conference) {
+            const phoneNumbers
+                = invitesLeftToSend.filter(({ type }) => type === 'phone');
+
+            // For each number, dial out. On success, remove the number from
+            // {@link invitesLeftToSend}.
+            const phoneInvitePromises = phoneNumbers.map(item => {
+                const numberToInvite = getDigitsOnly(item.number);
+
+                return conference.dial(numberToInvite)
+                     .then(() => {
+                         invitesLeftToSend
+                             = invitesLeftToSend.filter(invite =>
+                                 invite !== item);
+                     })
+                     .catch(error => logger.error(
+                         'Error inviting phone number:', error));
+            });
+
+            allInvitePromises = allInvitePromises.concat(phoneInvitePromises);
+        }
+
+        const usersAndRooms = invitesLeftToSend.filter(item =>
+            item.type === 'user' || item.type === 'room');
+
+        if (usersAndRooms.length) {
+            // Send a request to invite all the rooms and users. On success,
+            // filter all rooms and users from {@link invitesLeftToSend}.
+            const peopleInvitePromise = invitePeopleAndChatRooms(
+                inviteServiceUrl,
+                inviteUrl,
+                jwt,
+                usersAndRooms)
+                .then(() => {
+                    invitesLeftToSend = invitesLeftToSend.filter(item =>
+                        item.type !== 'user' && item.type !== 'room');
+                })
+                .catch(error => logger.error(
+                    'Error inviting people:', error));
+
+            allInvitePromises.push(peopleInvitePromise);
+        }
+
+        // Sipgw calls are fire and forget. Invite them to the conference
+        // then immediately remove them from {@link invitesLeftToSend}.
+        const vrooms = invitesLeftToSend.filter(item =>
+            item.type === 'videosipgw');
+
+        conference
+            && vrooms.length > 0
+            && dispatch(inviteVideoRooms(conference, vrooms));
```

Comment:
1. If `dispatch(inviteVideoRooms(conference, vrooms))` is to be in `sendInvitesForItems` and
2. given that `inviteVideoRooms` is an action creator from before this PR, then
3. `sendInvitesForItems` is logically an asynchronous action creator.

Am I missing something?